[integration tests flake]  fairsharing ./test/integration/singlecluster/scheduler/fairsharing

/kind flake

**What happened**:
failures on unreleated branches: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/6944/pull-kueue-test-integration-baseline-main/1970474873939562496

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/6942/pull-kueue-test-integration-baseline-main/1970475502288244736

**What you expected to happen**:
no failure
**How to reproduce it (as minimally and precisely as possible)**:

```

There were failures detected in the following suites:
  fairsharing ./test/integration/singlecluster/scheduler/fairsharing
Test Suite Failed
make: *** [Makefile-test.mk:72: test-integration] Error 1
```

--- a/pkg/cache/scheduler/fair_sharing.go
+++ b/pkg/cache/scheduler/fair_sharing.go
@@ -68,7 +68,12 @@ func (d DRS) PreciseWeightedShare() float64 {
 		// branches never reach here.
 		return math.Inf(1)
 	}
-	return d.unweightedRatio / d.fairWeight.AsFloat64Slow()
+	// We make a copy to ensure the fairWeight object doesn’t change after calling AsFloat64Slow().
+	// This can happen because quantity.AsDec mutates the underlying state
+	// (see https://github.com/kubernetes/apimachinery/blob/da5b06e2fb6698d6db8866899150ec2c1b4518d9/pkg/api/resource/quantity.go#L538).
+	// This mutation triggers the golang data race reporting when the DRS structure is used from multiple goroutines.
+	frWeightCopy := d.fairWeight.DeepCopy()
+	return d.unweightedRatio / frWeightCopy.AsFloat64Slow()
 }
 
 // CompareDRS compares two DRS values. A lower value

